src/components/ProjectSearch/TextSearch.js

```diff
@@ -49,6 +82,19 @@ export default class TextSearch extends Component {
     }
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.results !== this.props.results) {
+      if (this.focusedItem === null) {
+        return;
+      }
+      const { file, match } = this.focusedItem;
+
+      this.focusedItem[
+        file && !match ? "file" : "match"
+      ] = getMatchingFocusNode(nextProps.results, file || match);
+    }
+  }
+
   componentWillUnmount() {
```

Comment:
Nit: Same thing, `componentWillUnmount` is life cycle and shoud be above handers & renderers.